### PR TITLE
issue #2922

### PR DESCRIPTION
**Pull request by d3nwp**
## Basics

Check relevant points but **please do not remove entries**.
Do not describe the purpose of this PR in the PR description but:

- [x] Short descriptions should be in the release notes (added as entry in
      `doc/news/_preparation_next_release.md` which contains `_(my name)_`)
      **Please always add something to the the release notes.**
- [ ] Longer descriptions should be in documentation or in design decisions.
- [x] Describe details of how you changed the code in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, should be in the commit messages.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise and good to read
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the build server is happy** and also you
  say that everything is ready to be merged.

### CODE REVIEW

File: doc/GETSTARTED.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Get Started
+
+- Skill requirements
+ - Working with  Linux
```
Comment (markus2330) [inline doc/GETSTARTED.md]: MacOS and BSDs are also fully supported. Windows might be supported but it is not tested much.

File: doc/GETSTARTED.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ 
+   Dont panic! [GIT](https://git-scm.com/) is a distributed version constrol system to track changes of the source code in a project. We will use a single command of GIT to get 
+   the source code of Elektra. 
+ - Basic knowledge about make/cmake
+ 
+   Never heard about [make](https://www.gnu.org/software/make/) or [cmake](https://cmake.org/), it’s okey! We will use this build-management tools to compile, 
+   link dependencies etc. to build a executable program from the source code.  
+ - Furthermore, a programming/scripting language like C, C++, ruby, JavaScript etc.
+ 
+   You will need a programming language to fulfill your needs
```
Comment (markus2330) [inline doc/GETSTARTED.md]: ?

File: doc/GETSTARTED.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ 
+- Software requirements
+
+   We need to install some basic tools to run Elektra: cmake3 , git , and essential build tools (make, gcc, and some standard Unix tools; alternatively ninja and clang are
```
Comment (markus2330) [inline doc/GETSTARTED.md]: Spaces? cmake3 looks like distribution-specific.

File: doc/GETSTARTED.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+   ```
+- Installation
+
+  If you meet all of the software requirements you can get the source code of Elektra by using this command:
+  ```sh
+   git clone https://github.com/ElektraInitiative/libelektra.git 
+   ```
+   Run the following commands to compile Elektra with non-experimental parts where your system happens to fulfil the dependences:
+   ```sh
+   cd libelektra
```
Comment (markus2330) [inline doc/GETSTARTED.md]: Here is a lot of overlap to doc/COMPILE.md

File: doc/GETSTARTED.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+   
+    After you completed building Elektra on your own, you can execute these commands to install Elektra:
+    ```sh
+    sudo make install
+    sudo ldconfig
+   ```
+- Hello World!
+
+  Start with your very first Elektra application and follow these steps:
+  [Hello World!](doc/tutorials/hello-elektra.md)
```
Comment (markus2330) [inline doc/GETSTARTED.md]: Link broken, absolute link starts with /.

I am not sure if your tutorial covered everything that COMPILE.md covered?

**Comment by d3nwp** [comment]
sry for the merge request, I forgott to change the pr-settings in my github.

**Comment by d3nwp** [comment]
How can I avoid these "Merge remote-tracking branch 'upstream/master" commits...? Its so super annoying; I though I need to change the PR destination but it doesnt help. Do I need to "rebase"?
`git fetch upstream
git rebase upstream/master
git push origin master --force`

**Comment by markus2330** [comment]
> How can I avoid these "Merge remote-tracking branch 'upstream/master" commits...? Its so super annoying; I though I need to change the PR destination but it doesnt help. Do I need to "rebase"?

Yes, you need to rebase.

**Comment by sanssecours** [comment]
> ```
> git fetch upstream
> git rebase upstream/master
> git push origin master --force
> ```

If you want, you can also use a [single `pull` command to rebase](http://gitready.com/advanced/2009/02/11/pull-with-rebase.html) your current branch on the `master` branch of the remote `upstream`:

```sh
git pull --rebase upstream master
```

. Afterwards you still need to update your changes on the remote with `git push`, though.

Git also provides the option to [rebase by default](https://coderwall.com/p/tnoiug/rebase-by-default-when-doing-git-pull):

```sh
git config --global pull.rebase true
```

. That is also what I use, since I dislike merge commits. Merge commits make the `git` history much more complicated.

**Comment by markus2330** [comment]
@d3nwp Can you add this information (https://github.com/ElektraInitiative/libelektra/pull/3002#issuecomment-536864935)  in the docu? (and also everything else you noticed now what needs to be done for having a nice&clean PR.) The "How to create a PR" is still a big problem for newcomers.

**Comment by markus2330** [comment]
You can also link from .github/PULL_REQUEST_TEMPLATE.md to your "how to create a PR" so that the information will be found easily.

**Comment by d3nwp** [comment]
My PRs are auto-rejected on account of failing tests. I have changed only documentation parts and I am wondering how to get my PR to be accepted. I guess the tests are not mandatory to pass the review?!

**Comment by mpranj** [comment]
The tests are failing because the formatting check has failed. The formatting check is applied on documentation too, so it can fail even if you only changed documentation parts. Unfortunately cirrus-ci seems broken at the moment, so I can't say anything more.

Additionally, you need to add a line describing your contribution and your name to the upcoming release notes ([see jenkins output](https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/PR-3002/9/pipeline/316)). After you add this line, many other tests will be run on our jenkins instance, and it will tell you what went wrong with the formatting and how to fix it.

> I guess the tests are not mandatory to pass the review?!

Your PR will need to pass the tests before it can be merged, otherwise it breaks the builds for everyone.

**Comment by sanssecours** [comment]
> After you add this line, many other tests will be run on our jenkins instance, and it will tell you what went wrong with the formatting and how to fix it.

You can also see the formatting problems in the other build jobs. For example, to see why the build job `🍎 Clang` failed, you can use the steps given below.

1. Click on the “Details” link right beside the build job `🍎 Clang`. In the output on the right side below the header “Task Summary” you will see that the `tests` stage failed. The longer output below “Task Summary” shows that only the test `testscr_check_formatting` did not finish successfully.
2. Since that information does not tell you how you can fix the problem, please click on the link “View more details on Cirrus CI” at the bottom of the page.
3. Search for the term “testscr_check_formatting“ in the newly opened page. The test output is [below the second occurrence of the term](https://cirrus-ci.com/task/4881815048814592?command=tests#L336-L524).

There are multiple options to fix the problematic test.

1. You can use the instructions printed by the failing test.
2. Use the formatting commit added by Restyled (see issue #3003, commit message “Restyled by prettier-markdown“)
3. Format the updated files yourself, using one of the options given in the [coding guidelines](https://github.com/ElektraInitiative/libelektra/blob/master/doc/CODING.md#markdown-guidelines).

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -1,20 +1,43 @@
...
+
+Check the [Ideas](/doc/IDEAS.md) page if you are searching for a good topic to start with. You can also visit the [issue-tracker](https://github.com/ElektraInitiative/libelektra/issues). Do not hesitate to open a new issue if you want to ask a question, report or fix a bug or approach new topics.
+
+## Creating a Pull-Request
+
+If you want to publish your local changes to this project you have to create a new pull-request.
+
+1. Open your GitHub repository/fork
+2. In the tab *Codes* press the *New Pull-Request* button and choose a title and description
+3. Select the correct branches
```
Comment (markus2330) [inline .github/CONTRIBUTING.md]: This statement does not really help: where should be the own repo, and merge with what?

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -1,20 +1,43 @@
...
+
+## Creating a Pull-Request
+
+If you want to publish your local changes to this project you have to create a new pull-request.
+
+1. Open your GitHub repository/fork
+2. In the tab *Codes* press the *New Pull-Request* button and choose a title and description
+3. Select the correct branches 
+4. Commit and push your local changes in git (keep in mind to sync your fork - fetch, rebase & push) 
+5. Wait for the code-review
```
Comment (markus2330) [inline .github/CONTRIBUTING.md]: Pinging is also ok. And one should make sure that everything of .github/PULL_REQUEST_TEMPLATE.md is fulfilled. In particular the build server must be happy and useful docu provided, otherwise the code review will take unnecessarily long.

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -1,20 +1,43 @@
...
+
+## Code-Review
+
+After the pull-request your code will be reviewed and tested with some tests. If your pull-request passes the review , your changes will be merged to the branch.
```
Comment (markus2330) [inline .github/CONTRIBUTING.md]: " and tested with some tests" can be removed (might happen, might not, depends on the case). It will be merged to master.

**Comment by markus2330** [comment]
Please also make sure that all these documents properly link to each other.

**Comment by markus2330** [comment]
It seems like that the build server had a time-out during building. In such cases you can simply trigger a rebuild.

This is described in doc/BUILDSERVER.md. Could you please better integrate this information for newcomers?

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by d3nwp** [comment]
I am wondering why the Cirrus CI / 🔗 Check fails so often.. there is something wrong with broken links in some of the old release notes..!? Sometimes I have to commit my changes twice to overcome the test.

**Comment by sanssecours** [comment]
> I am wondering why the Cirrus CI / 🔗 Check fails so often.. there is something wrong with broken links in some of the old release notes..!? Sometimes I have to commit my changes twice to overcome the test.

Unfortunately our main Jenkins build server (that hosts `build.libelektra.org`) is very underpowered. That is the reason why the job `🔗 Check` will often be unable to access the links using the timeout of 5 seconds:

https://github.com/ElektraInitiative/libelektra/blob/f6fcf3b984c998a2d4633c85ccb9b5eccb7ba1e1/scripts/link-checker#L61-L69

. On the positive side, Djordje (Mistreated) and Dominic (Chemin1) are currently in the process of migrating Jenkins to a new more powerful server. Hopefully we do not have to restart `🔗 Check` this often after the upgrade.

> Sometimes I have to commit my changes twice to overcome the test.

You can also sign into Cirrus CI [here](https://cirrus-ci.com). After that you should be able to restart failing build jobs with a single click.

**Comment by sanssecours** [comment]
> It doesn't help ... I tried like 6-7 times. :(

That is not that often 😊. I had to [retry 25 times](https://cirrus-ci.com/task/5763776934051840) in PR #3065.

**Comment by d3nwp** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
Sorry for these problems. I hope @Mistreated will be able to migrate everything within the next week.

**Comment by markus2330** [comment]
> That is not that often blush. I had to retry 25 times in PR #3065.

@sanssecours isn't it possible to increase the timeout?

**Comment by sanssecours** [comment]
> @sanssecours isn't it possible to increase the timeout?

As far as I can tell, we just need to replace the timeout value of 5 seconds in `--timeout=5` with a higher value:

https://github.com/ElektraInitiative/libelektra/blob/f6fcf3b984c998a2d4633c85ccb9b5eccb7ba1e1/scripts/link-checker#L61-L69

.

**Comment by markus2330** [comment]
Thank you for the hint. I increased it in ece21a85b5d477399e7147eac56eb5ee706d6f90

**Comment by d3nwp** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by d3nwp** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
There is a conflicting file here, please rebase.

**Comment by d3nwp** [comment]
I resolved the conflicts and pushed the changes but I cant see the commit here. I checked my fork and in my fork the conflicts are resolved. 

I started the rebuild because the deploy process stopped with the error :"...manifest unknown..."

**Comment by markus2330** [comment]
> Yesterday I resolved the conflicts and pushed the changes but I cant see the commit here. I checked my fork and in my fork the conflicts are resolved.

Yes, looks good now.

> I started the rebuild because the deploy process stopped with the error :"...manifest unknown..."

Thank you! :sparkle:

File: .github/CONTRIBUTING.md [hunk 9]
```diff
@@ -1,20 +1,46 @@
...
+
+```sh
+$ git fetch upstream
+$ git rebase upstream/master
+$ git push origin master --force
+```
+
+## Issues
+
+Check the [Ideas](/doc/IDEAS.md) page if you are searching for a good topic to start with. You can also visit the [issue-tracker](https://github.com/ElektraInitiative/libelektra/issues). Do not hesitate to open a new issue if you want to ask a question, report or fix a bug or approach new topics.
```
Comment (Piankero) [inline .github/CONTRIBUTING.md]: You could also write about our labels, eg. "help wanted" or "Good first issue" (currently renamed "Hacktoberfest")

File: .github/CONTRIBUTING.md [hunk 10]
```diff
@@ -1,20 +1,46 @@
...
+
+## Creating a Pull-Request
+
+If you want to publish your local changes to this project you have to create a new pull-request.
+
+1. Open GitHub and navigate to _your_ libeketra-fork
```
Comment (Piankero) [inline .github/CONTRIBUTING.md]: typo in libelektra

File: .github/CONTRIBUTING.md [hunk 11]
```diff
@@ -1,20 +1,46 @@
...
+2. In the tab _Codes_ press the _New Pull-Request_ button and choose a title and fill in the [pull-request template](/.github/PULL_REQUEST_TEMPLATE.md)
+3. Now you should be able to set up the _Comparing changes_ settings. Choose the head repository and base repository. E.g.: if you have some changes in the master branch of your forked repository, select *head repository: yournickname/libelektra and *compare: master* and *base repositroy: ElektraInitiative/libelektra* and *base: master\*
+4. Add some information about the changes in the release notes (path of the file: /libelektra/doc/news/\_preparation_next_release.md), skipping this step may cause a rejected pull-request
```
Comment (Piankero) [inline .github/CONTRIBUTING.md]: to avoid frustration of the developer you should write about the formatting. The build server instantly fails if there is just have a single space too much at the end of the line (scripts\dev\reformat-all). This might require the contributor to either install docker (doc\tutorials\run_reformatting_script_with_docker.md) or the scripts itself but with additional tools installed (written in here doc\CODING.md).

File: .github/CONTRIBUTING.md [hunk 12]
```diff
@@ -1,20 +1,46 @@
...
+5. Commit and push your local changes in git (keep in mind to sync your fork - fetch, rebase & push)
+6. Wait for the code-review
+
+## Code-Review
+
+After the pull-request your code will be reviewed and if your pull-request passes the review , your changes will be merged to master.
```
Comment (Piankero) [inline .github/CONTRIBUTING.md]: After opening the pull-request, your code ....

Comment (Piankero) [inline .github/CONTRIBUTING.md]: passes review & build server requirements

Comment (Piankero) [inline .github/CONTRIBUTING.md]: You could also write that in many cases the build fails without the fault of the developer. (e.g., websites are down, build server stuck (as you can see yourself)). Some short section about these issues would be helpful. Also a link to the buildserver is helpful: doc\BUILDSERVER.md

File: doc/GETSTARTED.md [hunk 13]
```diff
@@ -0,0 +1,77 @@
+# Get Started
+
+- Skill requirements
+
+  - Operating system
+
+    We will need a unix-based operating system to run Elektra (Linux, BSD, macOS) but it's also possible to use Windows which is supported but not tested much. If you are new to the world of unix we recommend you to use one of the common linux distributions like [Ubuntu](https://ubuntu.com/download/desktop#download), [Linux Mint](https://www.linuxmint.com/download.php), [openSuse](https://software.opensuse.org/distributions/leap) or [Fedora](https://getfedora.org/en/).
```
Comment (Piankero) [inline doc/GETSTARTED.md]:
```suggestion
    We recommend a unix-based operating system to run Elektra (Linux, BSD, macOS) but it's also possible to use Windows which is supported but not yet fully tested.
```

File: doc/GETSTARTED.md [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+  - Using command-line interface and commands
+
+    The easiest way to compile, install and use Elektra is by using the terminal. We will introduce the basic commands which you will need to run Elektra for the very first time.
+
+  - Basic knowledge about git
+
+    Dont panic! [GIT](https://git-scm.com/) is a distributed version constrol system to track changes of the source code in a project. We will use a single command of GIT to get
```
Comment (Piankero) [inline doc/GETSTARTED.md]: I like this sort of writing but I think that those sentences here are not needed. If somebody wants to contribute to a repository on github he will for sure know about git. Just link git website in the "Basic knowledge about git"

File: doc/GETSTARTED.md (hunk 13 above)
Comment (Piankero) [inline doc/GETSTARTED.md]: If someone really wants to contribute to Elektra and does not have a unix based system, he will have severe problems learning unix based systems before actually contributing to Elektra. If someone already has experience with unix (which I assume the vast majority of Dev's contributing to C related projects have), these sentences are irrelevant. Therefore I would simply omit those sentences

File: doc/GETSTARTED.md [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+    the source code of Elektra.
+
+  - Basic knowledge about make/cmake
+
+    Never heard about [make](https://www.gnu.org/software/make/) or [cmake](https://cmake.org/), it’s okey! We will use this build-management tools to compile, link dependencies etc. to build a executable program from the source code.
```
Comment (Piankero) [inline doc/GETSTARTED.md]: If someone really did not hear about make or cmake he will also have troubles understanding "link dependencies" (Depite I know both tools for example I do not exactly know what happens on dependency linkage).

Comment (Piankero) [inline doc/GETSTARTED.md]:
```suggestion
    [make](https://www.gnu.org/software/make/) or [cmake](https://cmake.org/) are used to generate an executable program from the code. If you are not used to these tools its no problem, we will introduce them to you in latter sections.
```

File: doc/GETSTARTED.md [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+
+  - (optional) Furthermore, a programming/scripting language like C, C++, ruby, JavaScript etc. ...
```
Comment (Piankero) [inline doc/GETSTARTED.md]: I dont think that this is needed as it is obvious

File: doc/GETSTARTED.md [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+
+    ... if you are planning to contribute to Elektra.
+
+- Software requirements
+
+  We need to install some basic tools to run Elektra: cmake , git , and essential build tools (make, gcc, and some standard Unix tools; alternatively ninja and clang are also supported but not described here). Depending on your linux distribution use following commands to install these tools:
```
Comment (Piankero) [inline doc/GETSTARTED.md]: provide a link to the homepages of ninja clang gcc

File: doc/GETSTARTED.md [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+  sudo apt-get install cmake git build-essential
+  ```
+
+  Or on RPM (Red Hat Package Manager) based systems (like Fedora, openSUSE, CentOS etc.):
+
+  ```sh
+  sudo yum install -y cmake git gcc-c++
+  ```
+
+  Or on macOS Sierra, most of the build tools can be obtained by installing Xcode (from the App Store). Other required tools may be installed using [brew ](https://brew.sh/). First install brew as described on their website. Then issue the following command to get cmake to complete the basic requirements:
```
Comment (Piankero) [inline doc/GETSTARTED.md]: link to xcode website

File: doc/GETSTARTED.md [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+
+- Installation
+
+  If you meet all of the software requirements you can get the source code of Elektra by using this command:
+
+  ```sh
+  git clone https://github.com/ElektraInitiative/libelektra.git
+  ```
+
+  Run the following commands to compile Elektra with non-experimental parts where your system happens to fulfil the dependences:
```
Comment (Piankero) [inline doc/GETSTARTED.md]:
```suggestion
  Run the following commands to compile Elektra with non-experimental plugins where your system happens to fulfill the dependencies:
```

File: doc/GETSTARTED.md [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+
+  ```sh
+  cd libelektra  #navigate to libelektra
+  mkdir build  && cd build  #create and navigate to the build directory
+  cmake ..  # watch output to see if everything needed is included
+  ccmake .. # optional: overview of the available build settings (needs cmake-curses-gui)
```
Comment (Piankero) [inline doc/GETSTARTED.md]:
```suggestion
#  optionally run "ccmake .." to get an overview of the available build settings (needs cmake-curses-gui)
```

People copy paste such stuff (even though they shouldn't). This command interrupts the flow and might confuse those who are unfarmiliar with ccmake

File: doc/GETSTARTED.md [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+  make -j 5  # running makefile (compiling, linking etc.), -j specifies the number of jobs to run simultaneously
+  make run_nokdbtests  # optional: run tests
+  ```
+
+  With these commands you will be able to run the "Hello World!" example but usually you will need to use some of the plugins, tools and bindings of Elektra. Please take a look to the more detailed [compiling documentation](/doc/COMPILE.md). After you completed building Elektra on your own, you can execute these commands to install Elektra (please check the [installation documentation](/doc/INSTALL.md) if you are using Debian):
```
Comment (Piankero) [inline doc/GETSTARTED.md]: Is there a overview about all available plugins somewhere @markus2330 ? It would be nice to link it to the "plugins" keyword so users can have an impression of what actually exists and they might need

File: doc/GETSTARTED.md [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+
+  Run the following commands to compile Elektra with non-experimental parts where your system happens to fulfil the dependences:
+
+  ```sh
+  cd libelektra  #navigate to libelektra
+  mkdir build  && cd build  #create and navigate to the build directory
+  cmake ..  # watch output to see if everything needed is included
+  ccmake .. # optional: overview of the available build settings (needs cmake-curses-gui)
+  make -j 5  # running makefile (compiling, linking etc.), -j specifies the number of jobs to run simultaneously
+  make run_nokdbtests  # optional: run tests
```
Comment (Piankero) [inline doc/GETSTARTED.md]: Testing is a major pain for me and my system of elektra. Especially which commands have to be run as root in order to work correctly. I reconfigured my system build process (along with I think most developers here) to avoid such issues. You could link doc\TESTING.md in a sentence stating the difficulties on testing

File: doc/news/_preparation_next_release.md [hunk 23]
```diff
@@ -267,6 +267,9 @@ you up to date with the multi-language support provided by Elektra.
 - The [compile instructions](../COMPILE.md) do not assume that you use `make` or `gcc` to build Elektra anymore. _(René Schwaiger)_
 - Add hints about reformatting with docker. _(Dominic Jäger)_
 - Add instructions about sourcing on FreeBSD. _(Dominic Jäger)_
+- We updated links for the INI parsing library Nickel. _(René Schwaiger)_
```
Comment (Piankero) [inline doc/news/_preparation_next_release.md]: I think there was some merge mistakes that this sentence here is marked as diff

File: doc/GETSTARTED.md (hunk 21 above)
Comment (Piankero) [reply doc/GETSTARTED.md]: Maybe doc\BIGPICTURE.md?

**Comment by Piankero** [comment]
Thank you for the nice introduction!

I think it would be beneficial if there are some additional links (and possibly short explanations) of some basic concepts which any dev should be familiar with when wanting to contribute.

E.g., Key, KeySet, error handling, kdb commands, namespaces, arrays, error handling etc. (most of them are in the "tutorials page")

Additionally what I am missing is some sort of IDE stuff. Some hardcore guys like @markus2330 use Vim but for beginners I would not suggest that. We actually have a nice tutorial about how to use Clion with Elektra (doc\tutorials\contributing-clion.md) which helps tremendously. Also VSCode is very popular.

File: doc/GETSTARTED.md (hunk 15 above)
Comment (markus2330) [reply doc/GETSTARTED.md]: latter -> later?

File: doc/GETSTARTED.md (hunk 21 above)
Comment (markus2330) [reply doc/GETSTARTED.md]: The overview to plugins is in www.libelektra.org/plugins

File: .github/CONTRIBUTING.md (hunk 9 above)
Comment (markus2330) [reply .github/CONTRIBUTING.md]: I renamed it back to "good first issue" as October is finished.

File: .github/CONTRIBUTING.md (hunk 12 above)
Comment (markus2330) [reply .github/CONTRIBUTING.md]: Yes, we really should mention that in the docu, the build server problems are too often to not say it.

File: doc/GETSTARTED.md (hunk 14 above)
Comment (markus2330) [reply doc/GETSTARTED.md]: I also like the style. Maybe some students really do not know git, so I think it is okay to have it (maybe once in the whole tutorial)

File: doc/GETSTARTED.md (hunk 15 above)
Comment (markus2330) [reply doc/GETSTARTED.md]: If you will have troubles with cmake depends on what you want to do. If you only write a simple plugin, there should be no problem.

File: doc/GETSTARTED.md (hunk 16 above)
Comment (markus2330) [reply doc/GETSTARTED.md]: I think it is nice to also invite non-programmers to Elektra, e.g. for docu, website, translation,....

File: doc/GETSTARTED.md (hunk 21 above)
Comment (markus2330) [reply doc/GETSTARTED.md]: (link to /src/plugins in markdown)

File: doc/GETSTARTED.md (hunk 22 above)
Comment (markus2330) [reply doc/GETSTARTED.md]: I fully agree that there needs to be a good description of what needs to happen as root. Actually it is very simple: everything that writes to spec or system namespaces needs root. E.g. `kdb mount` writes to system, and because of that it needs root.

**Comment by markus2330** [comment]
@d3nwp can you please resolve the merge conflict and integrate the review comments?

**Comment by markus2330** [comment]
Unfortunately, you still did not rebase.

I now created a new PR  #3188 which might have created a conflict with this PR. It might be the best if you directly rebase on top of  #3188.

**Comment by Mistreated** [comment]
I rebased it on top of #3188.

**Comment by markus2330** [comment]
Something went wrong, #3188 contained commits from your PR, and not my commits (I initially pushed there) anymore. Seems like you force pushed the wrong commits to the wrong branch?

I force pushed my changes again to #3188.

**Comment by d3nwp** [comment]
Sorry, I was quite busy in the last couple days. I will check all the comments and changes on Thu.

**Comment by markus2330** [comment]
Thank you, it would be important that you take your time for this, the implementation task is essential.

**Comment by d3nwp** [comment]
jenkins build libelektra please

File: .github/CONTRIBUTING.md [hunk 24]
```diff
@@ -1,20 +1,49 @@
-# Documentation
+# Contributing
 
-We use a fork and pull model, you can read about
-[forking repos at github help](https://help.github.com/articles/fork-a-repo/).
+We use [GitHub](https://github.com/ElektraInitiative/libelektra/) to maintain this project. First of all you will need a GitHub account and [Git](https://www.git-scm.com/).
```
Comment (markus2330) [inline .github/CONTRIBUTING.md]: Git locally installed?

File: .github/CONTRIBUTING.md [hunk 25]
```diff
@@ -1,20 +1,49 @@
...
+If you are looking for a IDE you can use [CLion](https://www.jetbrains.com/clion/) ([setup tutorial](/doc/tutorials/contributing-clion.md)) or [Visual Studio Code](https://code.visualstudio.com/).
```
Comment (markus2330) [inline .github/CONTRIBUTING.md]: I would rephrase "you can use" as anybody is free to use anything. But instead something like "it is known to work" or "current Elektra developers use".

File: .github/CONTRIBUTING.md [hunk 26]
```diff
@@ -1,20 +1,49 @@
...
+
+```sh
+$ git fetch upstream
+$ git rebase upstream/master
+$ git push origin master --force
+```
+
+## Issues
+
+Check the [Ideas](/doc/IDEAS.md) page if you are searching for a good topic to start with. You can also visit the [issue-tracker](https://github.com/ElektraInitiative/libelektra/issues) and filter the list by pre-defined labels like _good first issue_ or _help wanted_.Do not hesitate to open a new issue if you want to ask a question, report or fix a bug or approach new topics.
```
Comment (markus2330) [inline .github/CONTRIBUTING.md]: link to good first issues and help wanted.

File: doc/GETSTARTED.md [hunk 27]
```diff
@@ -0,0 +1,77 @@
...
+
+    [make](https://www.gnu.org/software/make/) or [cmake](https://cmake.org/) are used to generate an executable program from the code. If you are not used to these tools its no problem, we will introduce them to you in later sections.
+
+  - We need also your skill set to improve Elektra
+
+    You can contribute to Elektra to improve the source code, website, documentation, translation etc.
+
+- Software requirements
+
+  We need to install some basic tools to run Elektra: cmake , git , and essential build tools (make, gcc, and some standard Unix tools; alternatively [ninja](https://ninja-build.org/) and [clang](https://clang.llvm.org/index.html) are also supported but not described here). Depending on your linux distribution use following commands to install these tools:
```
Comment (markus2330) [inline doc/GETSTARTED.md]:
```suggestion
  We need to install some basic tools to run Elektra: cmake, git, and essential build tools (make, gcc, and some standard Unix tools; alternatively [ninja](https://ninja-build.org/) and [clang](https://clang.llvm.org/index.html) are also supported but not described here). Depending on your linux distribution use following commands to install these tools:
```

**Comment by markus2330** [comment]
Thank you, great job! I am looking forward to a new PR soon!
